### PR TITLE
Fixups on some of @cdecker's plugins

### PR DESCRIPTION
**Pull request by cdecker**
Builds on top of #417

### CODE REVIEW

File: request-invoice/requestinvoice.py [hunk 1]
```diff
@@ -1,125 +0,0 @@
-#!/usr/bin/env python3
```
Comment (m-schmoock) [inline request-invoice/requestinvoice.py:1]: Why not moving this into the `/archive` like I did for the old `commando` plugin ( 748d50a ) instead?

File: backup/test_backup.py [hunk 2]
```diff
@@ -49,13 +49,17 @@ def test_start_no_init(node_factory, directory):
         options=opts, cleandir=False, may_fail=True, start=False
     )
 
-    with pytest.raises(ValueError):
```
Comment (m-schmoock) [inline backup/test_backup.py:52]: See https://github.com/lightningd/plugins/pull/416/commits/5892bdc13d27e4dceed41795bdae22845232be90 on how I solved that. 

The test still works correctly, we just need to wait for the 30secs timeout. Maybe we can change the test by setting the `wait_for` `TIMEOUT` value specically lower for this test.

Comment (m-schmoock) [reply backup/test_backup.py:52]: This commit is part of the PR https://github.com/lightningd/plugins/pull/416

File: request-invoice/requestinvoice.py (hunk 1 above)
Comment (cdecker) [reply request-invoice/requestinvoice.py:1]: Because it is rather annoyingly useless and pretty much requires a full rewrite anyway.

File: backup/test_backup.py (hunk 2 above)
Comment (cdecker) [reply backup/test_backup.py:52]: Merged that PR, now we need to rebase here :-)

**Comment by cdecker** [comment]
Rebased on top of `master` after applying #416

**Comment by m-schmoock** [comment]
Wow, actually green tests 🎆

**Comment by cdecker** [comment]
Nice, looks like things are passing now. @m-schmoock do you want to rebase #417 on top of `master` so and then I can merge the two back to back?

**Comment by cdecker** [comment]
By the way, further work I intend to do includes:

 - [ ] Downloading nightly and stable releases of CLN instead of building them
 - [ ] Test against stable and nightly daily, to have a canary for breaking plugins when we deprecate things
 - [ ] Split testing into separate jobs, to further isolate the runs, hopyfully we find a way to automatically file issues as well when a plugin fails, calling out the maintainers.

**Comment by m-schmoock** [comment]
> Nice, looks like things are passing now. @m-schmoock do you want to rebase #417 on top of `master` so and then I can merge the two back to back?

Done

File: backup/requirements.txt [hunk 3]
```diff
@@ -1,2 +0,0 @@
-pyln-client>=0.12
```
Comment (m-schmoock) [inline backup/requirements.txt:1]: Agree... But whats the trick to tell CLN to use poetry on a plugin?

The reason I'm still using pip requirements is that when I install them `--user` (and they don't clash with deps each other) CLN is able to start it without loading a special environment.

Maybe I'm missing information, if so we also need to update `README.md` on how to run plugins correctly with poetry. Currently when I have a poetry plugin I do `poetry export --without-hashes -f requirements.txt --output requirements.txt ` and install that again via `pip install --user -r requirement.txt`.  :D

File: noise/noise.py [hunk 4]
```diff
@@ -53,13 +53,27 @@ def to_dict(self):
...
+        if i < 2**8:
+            return struct.pack("!b", i)
+        elif i < 2**16:
+            return struct.pack('!H', i)
+        elif i < 2**32:
+            return struct.pack("!I", i)
+        else:
+            return struct.pack("!Q", i)
+
+    print("XXX", repr(scid))
```
Comment (m-schmoock) [inline noise/noise.py:67]: I see there is still debug output.
I repeat the review when your ready ;)

File: autopilot/lib_autopilot.py [hunk 5]
```diff
@@ -246,7 +246,7 @@ def __get_long_path_pdf(self, skew=True, smooth=False):
 
         s = sum(path_pdf.values())
         # FIXME: next line can raise a division by zero
```
Comment (m-schmoock) [inline autopilot/lib_autopilot.py:248]: This comment can be deleted I guess

**Review by m-schmoock — CHANGES_REQUESTED** [review]

File: backup/requirements.txt (hunk 3 above)
Comment (cdecker) [reply backup/requirements.txt:1]: Yeah, it's a bit tricky: if instead of a full she-bang pointing to a specific `python3` we use `/use/bin/env python3` then the plugin executable will use the `$PATH` envvar. Both `venv` and `poetry run` will prefix the `$PATH` with their `python3` location (allowing to enter and exit an env). So either `python3 -m pip install` or `python3 -m poetry install` will install things in an env. With these changes we get equivalence between the two, by disallowing poetry's virtualenv logic, and allocating a throwaway env for the test.

Long story short: use `python3 .ci/test.py [pluginname]` yo tun tests as they will be run on CI

**Comment by cdecker** [comment]
ACK https://github.com/lightningd/plugins/pull/418/commits/6887ecd347983206a75fcdec9085f37966a8fd6c

File: backup/requirements.txt (hunk 3 above)
Comment (m-schmoock) [reply backup/requirements.txt:1]: Thx. How does cln itself start a python plugin? Does it do a similar trick to check if it needs to load a special env or is it just running `python3 plugin.py` ?
